rcgen/src/certificate.rs

```diff
@@ -455,6 +455,24 @@ impl CertificateParams {
 		Ok(result)
 	}
 
+	/// Write a certificate's KeyUsage as defined in RFC 5280.
+	fn write_key_usage(&self, writer: DERWriter) {
+		// RFC 5280 defines 9 key usages, which we detail in our key usage enum
+		// We could use std::mem::variant_count here, but it's experimental
+		const KEY_USAGE_BITS: usize = 9;
+
+		if !self.key_usages.is_empty() {
```

Comment:
Let's invert this condition and do an early return instead?

Reply:
Done!